### PR TITLE
Stringify NSDates

### PR DESCRIPTION
**Pull request by nbarnold01**
I changed the behavior of `hyp_dictionary` to use an NSDateFormatter set to the ISO 8601 standard. There's also an added method `hyp_dictionaryWithDateFormatter` that accepts a custom date formatter.

### CODE REVIEW

File: Source/NSManagedObject+HYPPropertyMapper.m [hunk 1]
```diff
@@ -13,24 +13,24 @@ @implementation NSManagedObject (HYPPropertyMapper)
 
 - (void)hyp_fillWithDictionary:(NSDictionary *)dictionary {
     for (__strong NSString *key in dictionary) {
-
+
```
Comment (3lvis) [inline Source/NSManagedObject+HYPPropertyMapper.m]: Would you mind removing this trailing spaces? This utility might useful for you.

https://github.com/zenangst/MarvinXcode

File: Source/NSManagedObject+HYPPropertyMapper.m [hunk 2]
```diff
@@ -43,36 +43,45 @@ - (void)hyp_fillWithDictionary:(NSDictionary *)dictionary {
...
 
 - (NSDictionary *)hyp_dictionary {
-    NSMutableDictionary *managedObjectAttributes = [NSMutableDictionary new];
+    
+    return [self hyp_dictionaryWithDateFormatter:[self defaultDateFormatter]];
+}
 
+
+- (NSDictionary *)hyp_dictionaryWithDateFormatter:(NSDateFormatter *)formatter {
+
```
Comment (3lvis) [inline Source/NSManagedObject+HYPPropertyMapper.m]: Please drop the newlines after the method

File: Source/NSManagedObject+HYPPropertyMapper.m [hunk 3]
```diff
@@ -43,36 +43,45 @@ - (void)hyp_fillWithDictionary:(NSDictionary *)dictionary {
 }
 
 - (NSDictionary *)hyp_dictionary {
-    NSMutableDictionary *managedObjectAttributes = [NSMutableDictionary new];
+
```
Comment (3lvis) [inline Source/NSManagedObject+HYPPropertyMapper.m]: Please drop the newlines after the method declaration

**Comment by nbarnold01** [comment]
It looks like Travis failed because it's expecting NSDate objects to be returned instead of NSStrings. Perhaps the tests should be updated as well?

File: Source/NSManagedObject+HYPPropertyMapper.m [hunk 4]
```diff
@@ -104,18 +113,34 @@ - (NSDictionary *)hyp_dictionary {
...
+    
     return [managedObjectAttributes copy];
 }
 
+
+#pragma mark - Private
+
+- (NSDateFormatter *)defaultDateFormatter {
+    
+    //ISO 8601 standard
```
Comment (3lvis) [inline Source/NSManagedObject+HYPPropertyMapper.m]: Please drop the newlines after the method declaration

File: Source/NSManagedObject+HYPPropertyMapper.m [hunk 5]
```diff
@@ -104,18 +113,34 @@ - (NSDictionary *)hyp_dictionary {
...
+    
+    NSDateFormatter *dateFormatter = [[NSDateFormatter alloc] init];
```
Comment (3lvis) [inline Source/NSManagedObject+HYPPropertyMapper.m]: Building NSDateFormatters is performance heavy, you might want to wrap this one into a singleton.

Check **Re-Using Formatter Instances** in [NSHipter's article on NSFormatter](http://nshipster.com/nsformatter/)

**Comment by 3lvis** [comment]
@nbarnold01 Would you mind fixing/improving the unit tests, please?

This looks really good, thanks!

**Comment by 3lvis** [comment]
@nbarnold01 yeah, feel free to update the tests to represent the current format of returned dates on `hyp_dictionary`.

File: Source/NSManagedObject+HYPPropertyMapper.m (hunk 5 above)
Comment (nbarnold01) [inline Source/NSManagedObject+HYPPropertyMapper.m]: Ah, yes. Will do.

**Comment by nbarnold01** [comment]
I tried to follow the spacing guidelines. Let me know if it still needs some work.

File: Source/NSManagedObject+HYPPropertyMapper.m [hunk 6]
```diff
@@ -104,18 +110,35 @@ - (NSDictionary *)hyp_dictionary {
...
-
+    
     return [managedObjectAttributes copy];
 }
 
+
+#pragma mark - Private
+
+- (NSDateFormatter *)defaultDateFormatter {
+    //ISO 8601 standard
```
Comment (3lvis) [inline Source/NSManagedObject+HYPPropertyMapper.m]: Could you remove this comment, please?

**Comment by 3lvis** [comment]
@nbarnold01 Overall looks pretty good but I'm stilling finding some spaces, could you take a look at them?

screen shot 2015-09-04 at 13 28 56

This might help:

settings

**Comment by 3lvis** [comment]
:sparkles:  Great work @nbarnold01, you are fantastic! :sparkles:

**Comment by 3lvis** [comment]
@hyperoslo/ios-open-source This might be a breaking change for apps that relay on `hyp_dictionary` returning NSDates.

**Pull request merged**